File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        i = 0
        sm = 0
        n = len(nums)
        ans = n+1

        for j in range(n):
            sm+=nums[j]
            while(sm>=target):
                ans = min(ans,j-i+1)
                sm-=nums[i]
                i+=1
        return 0 if ans==n+1 else ans

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int i=0 , j=0,sum= 0;
        int n = nums.size();
        int ans = n+1;

        for(j=0;j<n;j++){
            sum = sum+nums[j];
            while(sum>=target){
                ans = min(ans,j-i+1);
                sum-=nums[i++];

            }
        }
        if(ans!=n+1){
            return ans;
        }
        else return 0;
    }
};